tests/data/test_sampling.py

```diff
@@ -126,3 +134,63 @@ def test_gpt_sample(seed, shuffle):
             # Check that the sequence is independent of `num_sample`.
             Assert.all_equal(samples, previous_samples[: len(samples)])
         previous_samples = samples
+
+
+@pytest.mark.skipif(not _extension_available, reason="CPP Extension not available")
+def test_build_padded_token_cumsum():
+    sizes = np.array([100, 256, 580, 600, 550, 89, 339, 430, 400, 795, 680, 50], dtype=np.int32)
+    sequence_length = 768
+    token_cumsum_rate = 4
+    offset = 0
+    # sequences with padding:
+    # [100, 256, 413 padded, 580, 189 padded, 600, 169 padded, 550, 89, 130 padded, 339, 430, 400, 369 padded, 680, 50, 39 padded]
+    # cumsums:
+    # [100, 356, 1349, 2307, 2857, 2946, 3415, 3845, 4245, 5294, 5344, 5383]
+    expected_cumsums = [0, 2307, 3845, 5383]
+    token_cumsum = build_padded_token_cumsum(sizes, sequence_length + 1, token_cumsum_rate, offset)
+    Assert.all_equal(token_cumsum, expected_cumsums)
+
+
+def get_test_seeds(num_seeds):
+    np.random.seed(42)
+    seeds = np.random.randint(0, num_seeds * 100, num_seeds)
+    return seeds.tolist()
+
+
+@pytest.mark.skipif(not _extension_available, reason="CPP Extension not available")
+def test_gpt_sample_padding():
+    for seed in get_test_seeds(10000):
```

Comment:
100 seeds should be more than enough. Anything more than that is just a waste of compute.